lib/shopify_app/app_proxy_verification.rb

```diff
@@ -0,0 +1,32 @@
+module ShopifyApp
+  module AppProxyVerification
+    extend ActiveSupport::Concern
+
+    # Usage:
+    #   include ShopifyApp::AppProxyVerification
+    #   before_action :verify_proxy_request
+    def verify_proxy_request
+      return head :unauthorized unless query_string_valid?(request.query_string)
+    end
```

Comment:
I think I understand now - using an `included do` block to add the before_action caused the tests to fail? It really should be done like this so can you post the error from your tests or I can check out the branch and we'll figure out a fix here.